packages/core/FuzzedDataProvider.ts

```diff
@@ -372,33 +372,100 @@ export class FuzzedDataProvider {
 	 * is not sufficiently long.
 	 * @param maxLength the maximum length of the string
 	 * @param encoding the encoding of the string
+	 * @param printable - a boolean, which defaults to false that indicates whether consumed strings
+	 * should be forced to contain only valid printable characters
 	 * @returns a `string` of length between 0 and `maxLength` (inclusive)
 	 */
 	consumeString(
 		maxLength: number,
-		encoding: BufferEncoding | undefined = "ascii"
+		encoding: BufferEncoding | undefined = "ascii",
+		printable: boolean | undefined = false
 	): string {
 		if (maxLength < 0) throw new Error("maxLength must be non-negative");
+		let result;
 		const arrayLength = Math.min(maxLength, this._remainingBytes);
-		const result = this.data.toString(
-			encoding,
-			this.dataPtr,
-			this.dataPtr + arrayLength
-		);
+
+		if (printable) {
+			result = this.bufToPrintableString(
+				this.data,
+				this.dataPtr,
+				this.dataPtr + arrayLength,
+				encoding
+			);
+		} else {
+			result = this.data.toString(
+				encoding,
+				this.dataPtr,
+				this.dataPtr + arrayLength
+			);
+		}
 		this.dataPtr += arrayLength;
 		this._remainingBytes -= arrayLength;
 		return result;
 	}
 
+	/**
+	 * Wrapping subtraction function that returns a number within a set limit
+	 * @param min is a number representing the minuend
+	 * @param sub is a number representing the subtrahend
+	 * @param lim sets the upper boundary
+	 * @returns a number n with the constraint 0 <= n < lim, as long as lim >=2
+	 */
+	wrappingSub(min: number, sub: number, lim: number): number {
+		if (lim < 2) {
+			throw new Error("Upper boundary should be greater than or equal to 2");
+		}
+		return (min - sub + lim) % lim;
+	}
+
+	/**
+	 * A simple conversion helper that yields the ASCII representaion of a character.
+	 * @param char is a single character from a string represented as a number
+	 * @returns a 'string' from a single character that was represented by a number
+	 */
+	toAscii(char: number): number {
+		if (char < 32 || char > 126) {
+			char = (this.wrappingSub(char, 32, 255) % 95) + 32;
+		}
+		return char;
+	}
+
+	/**
+	 * Helper function that converts the given string type into one that only
+	 * contains printable characters. This conversion produces seemingly arbitrary
```

Comment:
> This conversion produces seemingly arbitrary (non-smart) but printable strings, IFF the entropy of the input buffer is high enough.
I don't understand what this means. The graph above shows a skewed distribution. Maybe you can mention this. I think it's ok, because many of the functions in this class return skewed distributions.

I think it's important to mention that characters that were already printable will not be modified.